symengine/series_flint.cpp

```diff
@@ -0,0 +1,172 @@
+#include <symengine/series_flint.h>
+#include <symengine/series_visitor.h>
+
+#ifdef HAVE_SYMENGINE_FLINT
+namespace SymEngine {
+
+URatPSeriesFlint::URatPSeriesFlint(fp_t p, const std::string varname, const unsigned degree)
+        : SeriesBase(std::move(p), varname, degree) {
+
+}
+RCP<const URatPSeriesFlint> URatPSeriesFlint::series(const RCP<const Basic> &t, const std::string &x, unsigned int prec) {
+    fp_t p("2  0 1");
+    SeriesVisitor<fp_t, flint::fmpqxx, URatPSeriesFlint> visitor(p, x, prec);
+    return visitor.series(t);
+}
+
+unsigned int str_hash(const char* s, unsigned int seed = 0)
+{
+    unsigned int hash = seed;
+    while (*s)
+    {
+        hash = hash * 101  +  *s++;
+    }
+    return hash;
+}
+
+std::size_t URatPSeriesFlint::__hash__() const {
+    std::size_t seed = URATPSERIESPIRANHA;
+    hash_combine(seed, var_);
+    hash_combine(seed, degree_);
+    hash_combine(seed, str_hash(p_.to_string().c_str(), seed));
```

Comment:
So in https://github.com/symengine/symengine/blob/3f0c0947ae09bac9545334a6903d9d72c7043a52/symengine/symbol.cpp#L14 we use:
```c++
std::hash<std::string> str_hash;
hash_combine(seed, str_hash(p_.to_string()));
```
is there a reason to use a different hash here?